Week3/More JavaScript.js

```diff
@@ -0,0 +1,180 @@
+    if (age >= 4) {
+        age = "used";
+    } else {
+        age = "new";
+    }
+    console.log("a " + color + " " + age + " " + listOfVehicle[category]);
+}
+
+newVehicle("green", 3, 1);
+
+ 
+// 10. Use the list of vehicles to write an advertisement. So that it prints something like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.". (Hint: use a for loop.)
+
+listOfVehicle = ["car", "motorbike", "caravan", "bike"];
+
+let nameOfVehicle = "";
+
+for (i = 1; i < listOfVehicle.length; i++) {
+  if (i < listOfVehicle.length - 1) {
+    nameOfVehicle += ", " + listOfVehicle[i] + "s";
+  } else {
+    nameOfVehicle += " and " + listOfVehicle[i] + "s.";
+  }
+}
+
+console.log("Amazing Joe's Garage, we service " + listOfVehicle[0] + "s" + nameOfVehicle);
+
+// 11. What if you add one more vehicle to the list, can you have that added to the advertisement without changing the code for question 10?
+
+listOfVehicle = ["car", "motorbike", "caravan", "bike"];
+
+listOfVehicle.push("jeep");
+
+nameOfVehicle = '';
+
+for (i = 1; i < listOfVehicle.length; i++) {
+  if (i < listOfVehicle.length - 1) {
+    nameOfVehicle += ", " + listOfVehicle[i] + "s";
+  } else {
+    nameOfVehicle += " and " + listOfVehicle[i] + "s.";
+  }
+}
+
+console.log("Amazing Joe's Garage, we service " + listOfVehicle[0] + "s" + nameOfVehicle);
+  
+
+// 12. Create an empty object.
+
+    const class23 = new Object ();
+
+// 13. Create an object that contains the teachers that you have had so far for the different modules.
+// 14. Add a property to the object you just created that contains the languages that they have taught you.
+
+    class23.teachers = 'Philipp, Rob, Unmesh & Bonan, Yash';
+
+    class23.languages = 'Html, CSS, Git, JavaScript1'
+ 
+    console.log(class23); // {teachers: "Philipp, Rob, Unmesh & Bonan, Yash", languages: "Html, CSS, Git, JavaScript1"}
+
+// 15. Write some code to test two arrays for equality using == and ===. Test the following:
+
+    let x = [1, 2, 3];
+    let y = [1, 2, 3];
+    let z = y;
+
+    /*as we discussed the issue in the previous hw and lecture, they are objects and to ensure equlatiy, 
+    it needs to reference to the same memory locations */
+    
+    x == y //    false --> because of the different references.
+    x === y //   false --> because of the different references.
+    z == y //    true --> they reference to same memory locations.
+    z == x //    false --> there is no correlation.
+    
+// 16. 
+    const o1 = { foo: "bar" };
+    const o2 = { foo: "bar" };
+    const o3 = o2;
+
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+    
+    /* Changing o2 changes o3 since o3 is referenced to value of o2.
+    But changing o1 doesn`t change o3 since they are not correlated.*/
+
+// Does the order that you assign (o3 = o2 or o2 = o3) matter?
+    
+    /* It does matter since o2 is a constant and cannot be reassigned. */
```

Comment:
Ah, could you also show this by `console.log`'ing the results of making the changes?

Reply:
I added console.log for the results

Reply:
I added console.log for the results.